src/transaction/components/TransactionProvider.tsx

```diff
@@ -82,12 +105,59 @@ export function TransactionProvider({
     setLifeCycleStatus,
     transactionHashList,
   });
-  const { transactionHash, status: callStatus } = useCallsStatus({
+  // useSendCalls or useSendCall
+  // Used for contract calls with raw calldata.
+  const { status: statusSendCalls, sendCallsAsync } = useSendCalls({
     setLifeCycleStatus,
-    transactionId,
+    setTransactionId,
+  });
+  const {
+    status: statusSendCall,
+    sendCallAsync,
+    data: sendCallTransactionHash,
+  } = useSendCall({
+    setLifeCycleStatus,
+    transactionHashList,
   });
+
+  // Transaction Status
+  // For batched, use statusSendCalls or statusWriteContracts
+  // For single, use statusSendCall or statusWriteContract
+  const transactionStatuses = walletCapabilities.hasAtomicBatch
+    ? {
+        [TRANSACTION_TYPE_CALLS]: statusSendCalls,
+        [TRANSACTION_TYPE_CONTRACTS]: statusWriteContracts,
+      }
+    : {
+        [TRANSACTION_TYPE_CALLS]: statusSendCall,
+        [TRANSACTION_TYPE_CONTRACTS]: statusWriteContract,
+      };
+  const transactionStatus = transactionStatuses[transactionType];
```

Comment:
i would wrap this in a `useMemo` and include the lines above assuming `transactionStatuses` isn't used anywhere else